### PR TITLE
Index Initializer

### PR DESCRIPTION
**Pull request by DevonAleshire**
## Summary
Minor updates to Index initializer verbiage adding that Dictionary is still able to be initialized using collection initializer and focusing change on the fact that we are now using indexer allowing users to be more consistent with its actual usage/assignment.

Fixes #Issue_Number 5169

### CODE REVIEW

File: docs/csharp/whats-new/csharp-6.md [hunk 1]
```diff
@@ -508,12 +508,12 @@ if any, is lost.
 ## Index Initializers
 
 *Index Initializers* is one of two features that make collection
-initializers more consistent. In earlier releases of C#, you could use
-*collection initializers* only with sequence style collections:
+initializers more consistent with index usage. In earlier releases of C#, you could use
+*collection initializers* only with sequence style collections, including <xref:System.Collections.Generic.Dictionary%602> with braces around Key-Vlue pairs:
```
Comment (pkulikov) [inline docs/csharp/whats-new/csharp-6.md]: There is a typo and what about rephrasing: *...around Key-Vlue pairs* -> *...around key and value pairs*

File: docs/csharp/whats-new/csharp-6.md [hunk 2]
```diff
@@ -508,12 +508,12 @@ if any, is lost.
...
 
 [!code-csharp[ListInitializer](../../../samples/snippets/csharp/new-in-6/initializers.cs#ListInitializer)]
 
-Now, you can also use them with <xref:System.Collections.Generic.Dictionary%602> collections and similar types:
+Now, you can use them with <xref:System.Collections.Generic.Dictionary%602> collections and similar types more consistent regular index usage and assignment:
```
Comment (pkulikov) [inline docs/csharp/whats-new/csharp-6.md]: Please rephrase the sentence, it doesn't look grammatically correct now.

Comment (BillWagner) [reply docs/csharp/whats-new/csharp-6.md]: I'd rephrase as "and similar types. The new syntax supports assignment using an index into the collection."

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks for submitting this change @DevonAleshire We appreciate it.

I've looked at the changes, and I have a suggestion to one of the comments by @pkulikov.

If you can make an update to the same branch to update PR, I'll merge it.

Thanks again for your contribution.

File: docs/csharp/whats-new/csharp-6.md (hunk 2 above)
Comment (DevonAleshire) [reply docs/csharp/whats-new/csharp-6.md]: Updated and submitted with the changes. Sorry for the previous issues and look forward to learning more and assisting in a greater capacity with the documentation in the future! Thanks for the help. Let me know if there are any remaining changes to be made.

File: docs/csharp/whats-new/csharp-6.md (hunk 1 above)
Comment (pkulikov) [reply docs/csharp/whats-new/csharp-6.md]: @DevonAleshire thank you very much for updating the text.

**Comment by BillWagner** [comment]
These changes were incorporated by @DevonAleshire into PR #5577 Closing
